### DISCUSSION TITLE
unresolved external symbol

### DISCUSSION THREAD

**Discussion by Jeb1232** [jrouwe/JoltPhysics#1332] [Q&A] [answered]
Hi I'm facing an issue where jolt is failing to link to my project and I have tried everything I have made sure the runtime libraries match I have used the distribution config I have made sure i have put the correct folder in the "additional library directories" input. I just cannot wrap my head around as to why it is not working.

here is the output:
aPhysicsSystem.obj : error LNK2001: unresolved external symbol "bool (__cdecl* JPH::AssertFailed)(char const *,char const *,char const *,unsigned int)" (?AssertFailed@JPH@@3P6A_NPEBD00I@ZEA)

**Comment by jrouwe** [comment, marked as answer]
Most likely you're mixing Debug/Release builds. `AssertFailed` is gated behind the `JPH_ENABLE_ASSERTS` define, so most likely Jolt is built in Release mode (`NDEBUG` defined -> `JPH_ENABLE_ASSERTS` undefined) and your own application in Debug mode (`_DEBUG` defined -> `JPH_ENABLE_ASSERTS` defined). Including a Jolt header which uses `JPH_ASSERT` will cause this error.

**Reply by Jeb1232** [reply]
My project is in release mode and jolt is built in distribution mode and it still causes the error. I should also probably mention my project builds as a dll

**Reply by Jeb1232** [reply]
Ok I managed to fix it. I for some reason had defined JPH_ENABLE_ASSERTS in my project so I replaced it with NDEBUG and now it works
